src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -7518,11 +7519,11 @@ void CodeGen::genPutArgStkFieldList(GenTreePutArgStk* putArgStk)
                         case GT_CNS_INT:
                             if (fieldNode->IsIconHandle())
                             {
-                                inst_IV_handle(INS_push, fieldNode->AsIntCon()->gtIconVal);
+                                inst_IV_handle(INS_push, (target_ssize_t)fieldNode->AsIntCon()->gtIconVal);
```

Comment:
Same comment as below: "I don't think this is correct. For x64 hosted targeting x86, I presume gtIconVal will be a 64-bit pointer than needs to be relocated, but this will truncated it to 32 bits."